Source/Core/Core/PowerPC/Jit64/Jit_LoadStore.cpp

```diff
@@ -290,6 +292,75 @@ void Jit64::lXXx(UGeckoInstruction inst)
 	gpr.UnlockAllX();
 }
 
+void Jit64::dcbx(UGeckoInstruction inst)
+{
+	INSTRUCTION_START
+	JITDISABLE(bJITLoadStoreOff);
+
+	X64Reg addr = RSCRATCH;
+	X64Reg value = RSCRATCH2;
+	X64Reg tmp = gpr.GetFreeXReg();
+	gpr.FlushLockX(tmp);
+
+	if (inst.RA && gpr.R(inst.RA).IsSimpleReg() && gpr.R(inst.RB).IsSimpleReg())
+	{
+		LEA(32, addr, MRegSum(gpr.RX(inst.RA), gpr.RX(inst.RB)));
+	}
+	else
+	{
+		MOV(32, R(addr), gpr.R(inst.RB));
+		if (inst.RA)
+			ADD(32, R(addr), gpr.R(inst.RA));
+	}
+
+	MOV(32, R(value), R(addr));
+	SHL(32, R(value), Imm8(3));
+	SHR(32, R(value), Imm8(13));
+	MOV(64, R(tmp), ImmPtr(jit->GetBlockCache()->GetBlockBitSet()));
+	MOV(32, R(value), MComplex(tmp, value, SCALE_4, 0));
+
+	MOV(32, R(tmp), R(addr));
+	SHR(32, R(tmp), Imm8(5));
+	BT(32, R(value), R(tmp));
+
+	FixupBranch c = J_CC(CC_C, true);
+	SwitchToFarCode();
+	SetJumpTarget(c);
+	BitSet32 registersInUse = CallerSavedRegistersInUse();
+	ABI_PushRegistersAndAdjustStack(registersInUse, 0);
+	MOV(32, R(ABI_PARAM1), R(addr));
+	MOV(32, R(ABI_PARAM2), Imm32(32));
+	XOR(32, R(ABI_PARAM3), R(ABI_PARAM3));
+	ABI_CallFunction((void*)JitInterface::InvalidateICache);
+	ABI_PopRegistersAndAdjustStack(registersInUse, 0);
+	c = J(true);
+	SwitchToNearCode();
+	SetJumpTarget(c);
+
+	// dcbi
+	if (inst.SUBOP10 == 470)
+	{
+		TEST(16, M(&DSP::g_dspState), Imm16(1 << 9));
```

Comment:
Comment on what this is doing. Really non-obvious without looking at diff/history.